### PR TITLE
Tag/v5

### PR DESCRIPTION
**Pull request by victorjulien**
replaces #9793:
* addresses note comment
* adds a minor optimization to the tag implementation

### CODE REVIEW

**Review by jufajardini — APPROVED** [review]
Looks good to me!

File: src/detect-engine-tag.c [hunk 1]
```diff
@@ -493,6 +493,26 @@ static void TagHandlePacketHost(Host *host, Packet *p)
...
+    } else {
+        HostLock(p->host_src);
+    }
+    return p->host_src;
+}
+
+static Host *GetLockedDstHost(Packet *p)
+{
+    if (p->host_dst == NULL) {
+        p->host_dst = HostLookupHostFromHash(&p->dst);
```
Comment (inashivb) [inline src/detect-engine-tag.c:509]: Do we for this case and for the same case in `src` above not need to `HostDecrUsecnt` (facilitated by `HostRelease`) during the Unlock in the caller?

Comment (victorjulien) [reply src/detect-engine-tag.c:509]: No, the packet keeps the reference to the host(s) until it is freed/recycled. Then it calls `PacketReleaseRefs` to drop the references. This way other parts of the code that use the host can do so w/o having to do another (expensive) lookup.

**Comment by victorjulien** [comment]
Merged in #9816, thanks!

**Pull request closed without merging**